src/data/db.ts

```diff
@@ -0,0 +1,29 @@
+import Dexie, { Table } from 'dexie';
+import { Activity, Category, Day } from './interfaces';
+import {
+	ACTIVITY_STORE,
+	CATEGORY_STORE,
+	DAY_STORE,
+	StoreName,
+} from './interfaces/StoreNames';
+
+export class FeelingTrackerDatabase extends Dexie {
+	[DAY_STORE]!: Table<Day>;
+	[ACTIVITY_STORE]!: Table<Activity>;
+	[CATEGORY_STORE]!: Table<Category>;
+
+	constructor() {
+		super('FeelingTrackerDatabase');
+
+		const stores: { [storeName in StoreName]: string } = {
+			[DAY_STORE]: '++id, date, activities, rating',
+			[ACTIVITY_STORE]:
+				'++id,title,description,rating,startTime,endTime,iconId,categoryIds',
+			[CATEGORY_STORE]: '++id,name,color',
+		};
+
+		this.version(1).stores(stores);
+	}
+}
+
+export const db = new FeelingTrackerDatabase();
```

Comment:
A few issues here:
1. We only have a single DB in the entire app, so no need for the specific name. You can just name it Database.
2. When making a dedicated file for a class/interface/enum/component - the file name should always correcpond with the declaration. Rename to `Database.ts` accordingly.
3. To ensure this is a singleton, don't export the class declaration. You only need to export the instance. This way you make sure no other file can create a `new Database()`.
4. Another general rule of thumb for singletons: for testing purposes, always declare and export an interface with the corresponding properties to later inject into objects. In this case, your class `extends Dexie implements IDatabase`